File: IDSFintechPortal.Api/Models/Product.cs
namespace IDSFintechPortal.Api.Models
{
    public class Product
    {
        public int Id {get; set;}
        public  string Name {get; set;} = string.Empty;
        public string? Description {get; set;}
        public string? BusinessPurpose {get; set;}
        public int ProductStatusId { get; set; }
        public string? CurrentVersion { get; set; }
        public string? SupportedMarkets { get; set; }
        public string? Criticality { get; set; }
        public string? Technologies { get; set; }
        public string? Notes { get; set; }
    }
}

File: ids-fintech-portal-frontend/src/types/Product.ts
export interface Product {
    id: number;
  name: string;
  description: string | null;
  businessPurpose: string | null;
  productStatusId: number;
  currentVersion: string | null;
  supportedMarkets: string | null;
  criticality: string | null;
  technologies: string | null;
  notes: string | null;
}

export interface CreateProductDto {
    name: string;
  description?: string;
  businessPurpose?: string;
  productStatusId: number;
  currentVersion?: string;
  supportedMarkets?: string;
  criticality?: string;
  technologies?: string;
  notes?: string;
}

export interface UpdateProductDto extends CreateProductDto {}